drivers/clock_control/nrf_power_clock.c

```diff
@@ -1,280 +1,327 @@
-{
-	u32_t imask;
+			do_start =  (config->start_handler) ?
+					config->start_handler(dev) : true;
 
-	ARG_UNUSED(sub_system);
+			if (do_start) {
+				nrf_clock_event_clear(config->started_evt);
+				nrf_clock_task_trigger(config->start_tsk);
+				DBG(dev, "Triggered start task");
+			}
+		}
 
-	/* Test for started resource, if so, decrement reference and acquire
-	 * resource guard.
-	 */
-	imask = irq_lock();
+		/* if node is in the list it means that it is scheduled for
+		 * the second time.
+		 */
+		if (data) {
+			if (is_in_list(&clk_data->list, &data->node)) {
+				irq_unlock(key);
+				return -EALREADY;
+			}
 
-	if (!m16src_ref) {
-		irq_unlock(imask);
-		return -EALREADY;
-	}
+			sys_slist_append(&clk_data->list, &data->node);
+		}
 
-	if (--m16src_ref) {
-		irq_unlock(imask);
-		return -EBUSY;
+		irq_unlock(key);
 	}
 
-	if (m16src_grd) {
-		m16src_ref++;
-		irq_unlock(imask);
-		return -EAGAIN;
-	}
 
-	m16src_grd = 1U;
+	return 0;
+}
 
-	irq_unlock(imask);
+static int clock_start(struct device *dev, clock_control_subsys_t sub_system)
+{
+	return clock_async_start(dev, sub_system, NULL);
+}
 
-	/* re-entrancy and mult-context safe, and reference count is zero, */
+static void nrf_power_clock_isr(void *arg);
 
-	nrf_clock_task_trigger(NRF_CLOCK_TASK_HFCLKSTOP);
+static int clock_control_init(struct device *hfclk_dev)
+{
+	IRQ_CONNECT(DT_INST_0_NORDIC_NRF_CLOCK_IRQ_0,
+		    DT_INST_0_NORDIC_NRF_CLOCK_IRQ_0_PRIORITY,
+		    nrf_power_clock_isr, 0, 0);
 
-	/* release resource guard */
-	m16src_grd = 0U;
+	irq_enable(DT_INST_0_NORDIC_NRF_CLOCK_IRQ_0);
 
-	return 0;
-}
+	nrf_clock_lf_src_set(CLOCK_CONTROL_NRF_K32SRC);
 
-static int k32src_start(struct device *dev, clock_control_subsys_t sub_system)
-{
-	u32_t lf_clk_src;
-	u32_t imask;
-	u32_t stat;
-
-#if defined(CONFIG_CLOCK_CONTROL_NRF_K32SRC_BLOCKING)
-	u32_t intenset;
-#endif /* CONFIG_CLOCK_CONTROL_NRF_K32SRC_BLOCKING */
-
-	/* If the LF clock is already started, but wasn't initialized with
-	 * this function, allow it to run once. This is needed because if a
-	 * soft reset is triggered while watchdog is active, the LF clock will
-	 * already be running, but won't be configured yet (watchdog forces LF
-	 * clock to be running).
-	 *
-	 * That is, a hardware check won't work here, because even if the LF
-	 * clock is already running it might not be initialized. We need an
-	 * initialized flag.
-	 */
-
-	imask = irq_lock();
-
-	if (k32src_initialized) {
-		irq_unlock(imask);
-		goto lf_already_started;
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
+		nrf_clock_control_calibration_init(hfclk_dev);
 	}
 
-	k32src_initialized = 1U;
+	nrf_clock_int_enable((NRF_CLOCK_INT_HF_STARTED_MASK |
+			      NRF_CLOCK_INT_LF_STARTED_MASK |
+			      COND_CODE_1(CONFIG_USB_NRF52840,
+				(NRF_POWER_INT_USBDETECTED_MASK |
+				NRF_POWER_INT_USBREMOVED_MASK |
+				NRF_POWER_INT_USBPWRRDY_MASK),
+				(0))));
```

Comment:
Wouldn't such formatting:
```
	nrf_clock_int_enable(
		NRF_CLOCK_INT_HF_STARTED_MASK |
		NRF_CLOCK_INT_LF_STARTED_MASK |
		COND_CODE_1(CONFIG_USB_NRF52840,
			(NRF_POWER_INT_USBDETECTED_MASK |
			 NRF_POWER_INT_USBREMOVED_MASK |
			 NRF_POWER_INT_USBPWRRDY_MASK),
			(0)));
```
be a bit more readable?

Reply:
done